syntax = "proto3";

package minio_service;

service MinioService {
    rpc UploadFiles(FileUploadRequest) returns (FileUploadResponse);
    rpc UploadObject (UploadObjectRequest) returns (UploadObjectResponse);
}

message FileUploadRequest {
    string file_path = 1;
    string bucket_name = 2;
    string object_name = 3;
  }
  
message FileUploadResponse {
    bool success = 1;
    string message = 2;
  }

  message UploadObjectRequest {
    // Имя бакета, куда загружаем файл.
    string bucket_name = 1;
    // Имя объекта (файла).
    string object_name = 2;
    // Содержимое файла в виде строкового представления.
    string file_content = 3;
  }
  
  message UploadObjectResponse {
    // Успешно ли прошла загрузка.
    bool success = 1;
    // Сообщение об ошибке (если есть).
    string error_message = 2;
  }